openedx/stanford/cms/djangoapps/contentstore/tests/test_bulkupdate.py

```diff
@@ -0,0 +1,184 @@
+"""
+Unit tests for bulk update problem settings utility
+"""
+
+import ddt
+import mock
+
+from django.core import mail
+
+from xmodule.modulestore.django import modulestore
+from xmodule.modulestore.tests.factories import CourseFactory
+
+from cms.djangoapps.contentstore.tests.utils import CourseTestCase
+from contentstore.utils import reverse_course_url
+
+from ..views.utilities.bulkupdate import SHOW_ANSWER_OPTIONS
+from ..views.utilities.tasks import bulk_update_problem_settings
+
+
+@ddt.ddt
+class BulkUpdateTest(CourseTestCase):
+    """
+    Test for bulk update get and post methods
+    """
+    def setUp(self):
+        """
+        Creates the test course
+        """
+        super(BulkUpdateTest, self).setUp()
+        self.course = CourseFactory.create()
+        self.bulkupdate_url = reverse_course_url('utility_bulkupdate_handler', self.course.id)
+
+    def count_published_problems(self):
+        """
+        Returns the number of problems that are currently published in the course
+        """
+        count = 0
+        store = modulestore()
+        problems = store.get_items(
+            self.course.id,
+            qualifiers={"category": 'problem'},
+        )
+        for problem in problems:
+            if store.has_published_version(problem):
+                count += 1
+        return count
+
+    def test_get_bulkupdate_html(self):
+        """
+        Tests getting the HTML template and URLs for the bulkupdate page
+        """
+        response = self.client.get(self.bulkupdate_url, HTTP_ACCEPT='text/html')
+        self.assertContains(response, '/course/{}'.format(self.course.id))
+        self.assertContains(response, '/utility/bulkupdate/{}'.format(self.course.id))
+        # Verify dynamic content populates HTML correctly
+        self.assertContains(
+            response,
+            '<input class="input setting-input setting-input-number" type="number" value="0" min="0.0000" step="1">'
+        )
+        self.assertContains(response, '<option value="always" selected>always</option>')
+        for option in SHOW_ANSWER_OPTIONS[1:]:
+            self.assertContains(response, '<option value="{}">{}</option>'.format(option, option))
+
+    def test_bulkupdate_put_unsupported(self):
```

Comment:
We should combine all of the unsupported HTTP requests into a single, DDT-driven test.